MastermindLogic/src/main/java/nl/fhict/s3/mastermindlogic/GameEndpoint.java

```diff
@@ -111,19 +118,28 @@ private void cannotParseMessage(String message) {
     }
 
     private void registerGameOperation(Session session) {
-        Game game = new Game();
-        application.newGame(game);
+        Game game = application.getOpenGameOrNew();
+        int playerID = -1;
+        if(game.getPlayer1() == null) {
+            game.setPlayer1(new Player(0));
+            playerID = 0;
+        } else if(game.getPlayer2() == null) {
+            game.setPlayer2(new Player(1));
+            playerID = 1;
+        } else {
+            //TODO: error
+        }
+        
+        assert playerID != -1;
+
         games.put(game.getId(), new ArrayList<Session>());
         games.get(game.getId()).add(session);
         
         // Return gameID to user
-        Gson gson = new Gson();
-        WebSocketMessage returnMessage = new WebSocketMessage();
-        returnMessage.setGameId(game.getId());
-        returnMessage.setOperation(WebSocketMessageOperation.REGISTER_GAME);
-        returnMessage.setContent(gson.toJson(game.getId().toString()));
-        String jsonReturnMessage = gson.toJson(returnMessage);
-        session.getAsyncRemote().sendText(jsonReturnMessage);
+        sendMessage(session, game.getId(), WebSocketMessageOperation.REGISTER_GAME, game.getId().toString());
```

Comment:
waarom is de content game.getId().toString()? en playerId moet ook gevuld worden

Reply:
Omdat de content de gameID moet zijn in een string format en niet een UUID format....

Reply:
Die andere is wel een goed punt